lib/model/model.go

```diff
@@ -1243,18 +1243,65 @@ func (m *Model) updateLocals(folder string, fs []protocol.FileInfo) {
 		return
 	}
 	files.Update(protocol.LocalDeviceID, fs)
-
+	
 	filenames := make([]string, len(fs))
 	for i, file := range fs {
 		filenames[i] = file.Name
 	}
-
+	
 	events.Default.Log(events.LocalIndexUpdated, map[string]interface{}{
 		"folder":    folder,
 		"items":     len(fs),
 		"filenames": filenames,
 		"version":   files.LocalVersion(protocol.LocalDeviceID),
 	})
+	
+	// Lets us know if file/folder change was originated locally or from a network
+	// sync update.  Now write these to a global log file.
+	if remoteUpdate != true {
+		m.localDiskUpdate(m.folderCfgs[folder].Path(), fs)
+	}
+}
+
+func (m *Model) localDiskUpdate(path string, files []protocol.FileInfo) {
+	// For windows paths, strip unwanted chars from the front
+	path = strings.Replace(path, "\\\\?\\", "", 1) 
+	
+	for _, file := range files {
+		objType := "file"
+		action := "modified"
+		
+		for _, version := range file.Version {
+			if m.shortID == version.ID {
+				// If our local vector is verison 1 AND it is the only version vector so far seen for this file then
+				// it is a new file.  Else if it is > 1 it's not new, and if it is 1 but another shortId version vector
+				// exists then it is new for us but created elsewhere so the file is still not new but modified by us.
+				// Only if it is truely new do we change this to 'added', else we leave it as 'modified'.
+				if version.Value == 1 && len(file.Version) == 1 {
+					action = "added"
+				}
+			}
+		}
+		
+		if file.IsDirectory() { objType = "dir" } // if dir change 'file' to 'dir'
+		if file.IsDeleted() { action = "deleted" } // if deleted change 'added/modified' to 'deleted'
+		
+		// If the file is a level or more deep then the forward slash seperator is embedded
+		// in the filename and makes the path look wierd on windows, so lets fix it
+		filename := file.Name
+		if runtime.GOOS == "windows" {
+			filename = strings.Replace(file.Name, "/", string(filepath.Separator), -1)
+		}
+		// And append it to the filepath
+		filepath := path + string(filepath.Separator) + filename
+		
+		events.Default.Log(events.LocalDiskUpdated, map[string]interface{}{
+		"devicename": m.deviceName,
+		"action":     action,
+		"objecttype": objType,
```

Comment:
Lets call these just `type` and `path`. We don't refer to "objects" anywhere else, and the file might be a directory of course.